lib/src/neu/generators/json-schema/json-schema-type-util.ts

```diff
@@ -0,0 +1,200 @@
+import assertNever from "assert-never";
+import {
+  areBooleanLiteralTypes,
+  areFloatLiteralTypes,
+  areIntLiteralTypes,
+  areStringLiteralTypes,
+  isBooleanLiteralType,
+  isFloatLiteralType,
+  isIntLiteralType,
+  isNotLiteralType,
+  isStringLiteralType,
+  Type,
+  TypeKind
+} from "../../types";
+import { JsonSchemaObject, JsonSchemaType } from "./json-schema-specification";
+
+/**
+ * Generate a JSON Schema type definition. `objectAdditionalProperties` may
+ * be used to configure whether additional properties should be allowed on
+ * object types. This can be useful for data validation purposes where
+ * property validation can be strict or lenient.
+ *
+ * @param type a contract type
+ * @param objectAdditionalProperties whether to allow additional properties for objects
+ */
+export function typeToJsonSchemaType(
+  type: Type,
+  objectAdditionalProperties: boolean = true // TODO: expose proper configuration object
```

Comment:
Did you create a ticket for it ? Or you can do it now if it doesn't too much time.

Reply:
No I haven't, this was existing code. There's no real reason immediately to expose a configuration object until there's more than one config option. It serves as a reminder for when someone decides to add more things.